### ISSUE TITLE
False EANSIBLE0016 when checking if task NOT changed

### ISSUE THREAD

**Issue by strangeman** [ansible/ansible-lint#258]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.3.0.0
ansible-lint 3.4.12
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

```
  - name: backup server configs
    win_shell: cp C:\ProgramData\App\{{ app_name }}.config C:\ProgramData\App\{{ app_name }}.config{{ backup_postfix }}
    args:
      removes: C:\ProgramData\App\{{ app_name }}.config

  - name: copy server configs
    win_template:
      src: "{{ deploy_env }}/apps/{{ app_name }}.xml.j2"
      dest: "C:\ProgramData\App\{{ app_name }}.config"
    register: app_config

  - name: delete backup because config file wasn't changed
    win_file:
      path: C:\ProgramData\App\{{ app_name }}.config{{ backup_postfix }}
      state: absent
    when: not (app_config|changed)
```
Third task shouldn't raise `[EANSIBLE0016] Tasks that run when changed should likely be handlers`, because there is nothing like handlers logic

# Actual Behaviour (Bug report only)

Ansible-lint show error `[EANSIBLE0016] Tasks that run when changed should likely be handlers`

**Comment by vegai** [comment]
Possibly fixed by https://github.com/willthames/ansible-lint/pull/253

**Comment by willthames** [comment]
#253 won't fix this.

The problem here is that ANSIBLE0016 is a bit stupid (I can say that as I wrote it), and looks for `|changed` or `.changed` in a `when` condition, but takes no account of a `not`.

I'm not sure how best to fix this - the test is a heuristic at best, and adding more hacks on top (e.g. to check if `not` appears too) is probably not the best approach.
